pkg/component/gardenerapiserver/hvpa.go

```diff
@@ -0,0 +1,167 @@
+						MinReplicas: pointer.Int32(replicas),
+						MaxReplicas: maxReplicas,
+						Metrics: []autoscalingv2beta1.MetricSpec{
+							{
+								Type: autoscalingv2beta1.ResourceMetricSourceType,
+								Resource: &autoscalingv2beta1.ResourceMetricSource{
+									Name:                     corev1.ResourceCPU,
+									TargetAverageUtilization: pointer.Int32(80),
+								},
+							},
+							{
+								Type: autoscalingv2beta1.ResourceMetricSourceType,
+								Resource: &autoscalingv2beta1.ResourceMetricSource{
+									Name:                     corev1.ResourceMemory,
+									TargetAverageUtilization: pointer.Int32(80),
+								},
+							},
+						},
+					},
+				},
+			},
+			Vpa: hvpav1alpha1.VpaSpec{
+				Selector: &metav1.LabelSelector{MatchLabels: vpaLabels},
+				Deploy:   true,
+				ScaleUp: hvpav1alpha1.ScaleType{
+					UpdatePolicy: hvpav1alpha1.UpdatePolicy{
+						UpdateMode: pointer.String(hvpav1alpha1.UpdateModeAuto),
+					},
+					StabilizationDuration: pointer.String("3m"),
+					MinChange: hvpav1alpha1.ScaleParams{
+						CPU: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("300m"),
+							Percentage: pointer.Int32(80),
+						},
+						Memory: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("200M"),
+							Percentage: pointer.Int32(80),
+						},
+					},
+				},
+				ScaleDown: hvpav1alpha1.ScaleType{
+					UpdatePolicy: hvpav1alpha1.UpdatePolicy{
+						UpdateMode: pointer.String(hvpav1alpha1.UpdateModeAuto),
+					},
+					StabilizationDuration: pointer.String("15m"),
+					MinChange: hvpav1alpha1.ScaleParams{
+						CPU: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("600m"),
+							Percentage: pointer.Int32(80),
+						},
+						Memory: hvpav1alpha1.ChangeParams{
+							Value:      pointer.String("600M"),
+							Percentage: pointer.Int32(80),
+						},
+					},
+				},
+				LimitsRequestsGapScaleParams: hvpav1alpha1.ScaleParams{
+					CPU: hvpav1alpha1.ChangeParams{
+						Value:      pointer.String("1"),
+						Percentage: pointer.Int32(70),
+					},
+					Memory: hvpav1alpha1.ChangeParams{
+						Value:      pointer.String("1G"),
+						Percentage: pointer.Int32(70),
+					},
+				},
+				Template: hvpav1alpha1.VpaTemplate{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: vpaLabels,
+					},
+					Spec: hvpav1alpha1.VpaTemplateSpec{
+						ResourcePolicy: &vpaautoscalingv1.PodResourcePolicy{
+							ContainerPolicies: []vpaautoscalingv1.ContainerResourcePolicy{{
+								ContainerName: containerName,
+								MinAllowed: corev1.ResourceList{
+									corev1.ResourceMemory: resource.MustParse("400M"),
+								},
+								MaxAllowed: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("4"),
+									corev1.ResourceMemory: resource.MustParse("25G"),
+								},
+							}},
+						},
+					},
+				},
+			},
+			WeightBasedScalingIntervals: []hvpav1alpha1.WeightBasedScalingInterval{{
+				VpaWeight:         hvpav1alpha1.HpaOnly,
```

Comment:
Do we not need this handling?
https://github.com/gardener/gardener/blob/06f3b19981f6ee4fdfb2471b87a1f2cb5bb9f84e/charts/gardener/controlplane/charts/runtime/templates/apiserver/hvpa.yaml#L83

Reply:
In our case, we max replicas is hard-coded to 4, hence this condition is always true :)